File: A1/A1-067/A1-067.cpp
#include <bits/stdc++.h>
using namespace std;

string is_member;
int item_count;
double total_price = 0.0;

int main () {
    cin >> is_member;
    
    cin >> item_count;
    for (int i = 0; i < item_count; i++) {
        double price;
        cin >> price;
        total_price += price;
    }

    if (is_member == "Y") {
        total_price *= 0.95;
    }

    if (is_member == "N" && total_price >= 500) {
        total_price *= 0.97;
    }

    cout << fixed << setprecision(2);
    cout << round(total_price * 100.0) / 100.0 << endl;
}

File: A1/A1-067/A1-067.py
# --------------------------------------------------
# File Name : A1-067.py
# Problem   : Member Cost
# Author    : Worralop Srichainont
# Date      : 2026-01-03
# --------------------------------------------------

# Initialize membership status and total price
is_member = False
total_price = 0.0

# Read membership status from input
data = input().strip()
if data == "Y":
    is_member = True

# Read number of items and their prices
items_count = int(input().strip())
for _ in range(items_count):
    total_price += float(input().strip())

# Apply discounts based on membership status and total price
if is_member:
    total_price *= 0.95
elif not is_member and total_price >= 500:
    total_price *= 0.97

# Output the final total price rounded to two decimal places
print(f"{total_price:.2f}")
